scripts/add-application-network-policies.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -e
+
+if [ "$#" -lt 3 -o "$#" -gt 4 ]; then
+    echo "Usage: ./scripts/add-application-network-policies.sh <vars_yaml> <base_yaml> <application_name> [<application_suffix>]"
+    exit 1
+fi
+
+export VARS_YAML=$1
+export BASE_YAML=$2
+export APPLICATION_NAME=$3
+export APPLICATION_SUFFIX=$4
+
+# In each of the cases for ingress and egress applications, we first apply the policy to any instances of the (in|e)gress
+# application that might (also?) be in the process of being released. The order that the two commands are run in *is* relevant
+# and designed to not to allow any policy combinations to be missed even in the face of concurrent release processes. As such,
+# we must be tolerant to any of these commands failing because any of these application names may or may not exist at the exact
+# point in time the command is executed. This is ok, because the *other* release process is also expected to be attempting
+# to create these policies at the same time and this routine is designed so that at least one of them should be able to succeed
+# before an application is started.
+#
+# And of course, because of this the commands also need to be robust against the network policy already existing
+
+for EGRESS_APPLICATION_NAME in $(yq -rs '.[0] * .[1] | .[env.APPLICATION_NAME]?.egress_to_applications?[]?' "$BASE_YAML" "$VARS_YAML") ; do
+    echo "Adding policy for access from ${APPLICATION_NAME}${APPLICATION_SUFFIX} to $EGRESS_APPLICATION_NAME-release..."
+    cf add-network-policy ${APPLICATION_NAME}${APPLICATION_SUFFIX} --destination-app $EGRESS_APPLICATION_NAME-release --protocol tcp --port 8080 || echo "...but that's probably fine."
```

Comment:
If we can assume that `$EGRESS_APPLICATION_NAME-release` is always going to have suffix `-release`, can we assume that for the `${APPLICATION_NAME}${APPLICATION_SUFFIX}` too?

Reply:
I made `$APPLICATION_SUFFIX` customizable _really_ so that this script could double up for the `add-all-app-network-policies` command, which will need to operate for the actual target application name - in this case `$APPLICATION_SUFFIX` can be left blank.

Reply:
(whereas we always need to consider the possibility of the "other party" app being in the middle of a release, so operate against both possibilities)

Reply:
Cool - makes sense 👍